pkg/vfs/archive.go

```diff
@@ -0,0 +1,62 @@
+package vfs
+
+import (
+	"archive/zip"
+	"fmt"
+	"io"
+	"net/http"
+	"path/filepath"
+)
+
+// ZipMime is the content-type for zip archives
+const ZipMime = "application/zip"
+
+// Archive is the data to create a zip archive
+type Archive struct {
+	Name  string   `json:"name"`
+	Files []string `json:"files"`
+}
+
+// Serve creates on the fly the zip archive and streams in a http response
+func (a *Archive) Serve(c Context, w http.ResponseWriter) error {
+	header := w.Header()
+	header.Set("Content-Type", ZipMime)
+	header.Set("Content-Disposition", "attachment; filename="+a.Name+".zip")
```

Comment:
I'd expect golang `net/http` to escape `\n`s and `\r`s from headers but this is worth a check (and commentary) and it makes me wonder if we shouldn't sanitize this input.